packages/react-kit/src/utils/with-rx2.ts

```diff
@@ -44,11 +42,13 @@ export const withRX = <P extends object>(Target: ComponentType<P>) => <D extends
 					props[key].pipe(map((value: unknown) => ({ [key]: value }))),
 				);
 				this.inputSubscription = merge(...inputs)
-					.pipe(observeOn(scheduler))
+					.pipe(options.scheduler ? observeOn(options.scheduler) : identity)
```

Comment:
I think this code path is hot enough not to use `pipe` at all when we don't have a custom scheduler